timetracker/entries/forms.py

```diff
@@ -39,11 +51,15 @@ def clean(self):
         # is preserved
         cleaned_data = super(EntryForm, self).clean()
 
-        # Get the start and end values from the cleaned_data dictionary, or None
+        # Get the start and stop values from the cleaned_data dictionary, or None
         # if the dictionary keys are missing
         start = cleaned_data.get('start', None)
-        end = cleaned_data.get('end', None)
+        stop = cleaned_data.get('stop', None)
+        if 'create_endnow' in self.data:
+            stop = timezone.now()
+            cleaned_data['stop'] = stop
```

Comment:
Given the intended task (see entries.html) this isn't too bad. I'm wouldn't know of any particular pattern people use to accomplish this kind of behavior.